src/test/fuzz/txdownloadman.cpp

```diff
@@ -0,0 +1,426 @@
+            },
+            [&] {
+                TxValidationState state;
+                state.Invalid(fuzzed_data_provider.PickValueInArray(TESTED_TX_RESULTS), "");
+                txdownload_impl.MempoolRejectedTx(rand_tx, state, rand_peer, fuzzed_data_provider.ConsumeBool());
+            },
+            [&] {
+                GenTxid gtxid = fuzzed_data_provider.ConsumeBool() ?
+                                GenTxid::Txid(rand_tx->GetHash()) :
+                                GenTxid::Wtxid(rand_tx->GetWitnessHash());
+                txdownload_impl.AddTxAnnouncement(rand_peer, gtxid, time, /*p2p_inv=*/fuzzed_data_provider.ConsumeBool());
+            },
+            [&] {
+                const auto getdata_requests = txdownload_impl.GetRequestsToSend(rand_peer, time);
+                // TxDownloadManager should not be telling us to request things we already have.
+                // Exclude m_lazy_recent_rejects_reconsiderable because it may request low-feerate parent of orphan.
+                for (const auto& gtxid : getdata_requests) {
+                    Assert(!txdownload_impl.AlreadyHaveTx(gtxid, /*include_reconsiderable=*/false));
+                }
+            },
+            [&] {
+                const auto& [should_validate, maybe_package] = txdownload_impl.ReceivedTx(rand_peer, rand_tx);
+                // The only possible results should be:
+                // - Don't validate the tx, no package.
+                // - Don't validate the tx, package.
+                // - Validate the tx, no package.
+                // The only combination that doesn't make sense is validate both tx and package.
+                Assert(!(should_validate && maybe_package.has_value()));
+                if (should_validate) {
+                    Assert(!txdownload_impl.AlreadyHaveTx(GenTxid::Wtxid(rand_tx->GetWitnessHash()), /*include_reconsiderable=*/true));
+                }
+                if (maybe_package.has_value()) {
+                    CheckPackageToValidate(*maybe_package, rand_peer);
+
+                    const auto& package = maybe_package->m_txns;
+                    // Parent is in m_lazy_recent_rejects_reconsiderable and child is in m_orphanage
+                    Assert(txdownload_impl.RecentRejectsReconsiderableFilter().contains(rand_tx->GetWitnessHash().ToUint256()));
+                    Assert(txdownload_impl.m_orphanage.HaveTx(maybe_package->m_txns.back()->GetWitnessHash()));
+                    // Package has not been rejected
+                    Assert(!txdownload_impl.RecentRejectsReconsiderableFilter().contains(GetPackageHash(package)));
+                    // Neither is in m_lazy_recent_rejects
+                    Assert(!txdownload_impl.RecentRejectsFilter().contains(package.front()->GetWitnessHash().ToUint256()));
+                    Assert(!txdownload_impl.RecentRejectsFilter().contains(package.back()->GetWitnessHash().ToUint256()));
```

Comment:
I believe it's very rare based on our filter parameters. Each generation has 60k entries expected, and according to my debug-fu has 20 hash functions, and 21,56,646 filter bits. Since the harness does up to 10k iterations(let's assume each iteration makes a unique tx), that is 10k transactions, this leads to:

```
P_fp = (1 - e^-(20*10000/2156646))^20
        ~= 8.8183022e-22
```

This assumes independence of probabilities of each bit being set. Once we get up to 60k iterations, assuming they are all added to the filter, then FP rate will hit ~1/1M. If we set total iterations to 1.5k, then we drop below cryptographic safety margins of 2^-128.